File: Projekt/Frontend/system-kliniki-medycznej/src/app/features/patient-disease/create-patient-disease-request.ts
import {HospitalizationInfo} from '../patient-card/model/hospitalization-info';
import {MedicineDto} from '../medicine/model/medicine-dto';
import {DiseaseDto} from '../disease/model/disease-dto';

export interface CreatePatientDiseaseRequest {
  patientCardId: string;
  diseaseDto: DiseaseDto;
  doctorId: number;
  description: string;
  hospitalization: {
    medicine: MedicineDto,
    dosage: string,
    notes: string
  }[];
}

File: Projekt/Backend/system-kliniki-medycznej/src/main/java/org/back/systemklinikimedycznej/patient/controllers/dto/CreatePatientDiseaseRequest.java
package org.back.systemklinikimedycznej.patient.controllers.dto;

import org.back.systemklinikimedycznej.disease.dto.DiseaseDto;

import java.util.List;
import java.util.UUID;

public record CreatePatientDiseaseRequest(
        UUID patientCardId,
        DiseaseDto diseaseDto,
        Long doctorId,
        String description,
        List<CreateHospitalizationRequest> hospitalizations
) {
}
